### PR TITLE
Add `sync-pr-commit-title` feature (replaces `fix-squash-and-merge-title`)

### PR DESCRIPTION
**Pull request by bfred-it**
Fixes https://github.com/sindresorhus/refined-github/issues/1790
Closes https://github.com/sindresorhus/refined-github/issues/1743

- When the commit title is manually updated, RGH will offer to also update the PR title
- Title/message features merged into one
- The title box is focused after it's opened
- The message box now expands to fit the content
- This now applies to Merge commits too

# Next

- [x] Readme
- [x] Add migration in background.js
- [x] Should we drop “fix-squash-and-merge-message”? I'm not entirely sure the new message is useful anyway and it contains a lot of noise ("fixes this and that", "I tried this", "What do you think", PR template comment, etc)
- [x] Does the "PR title update" part belong to the same feature? There's some overlap and both features kinda do the same thing, in two directions

# Test 

<details>
<summary>Commit info update</summary>

1. Visit PR with ONE commit (with 2+ commits, GitHub already updates **the title**)
2. Open the merge box (as Squash or Merge)
---
1. Visit PR
2. Change PR title
2. Open the merge box (as Squash or Merge)

</details>

<details>
<summary>PR title update</summary>

1. Visit PR
2. Open the merge box (as Squash or Merge)
3. Change title (the message will appear)
4. Undo changes (the message will disappear)
5. Merge (the PR title will be updated if the message was shown)
---
1. Visit PR
2. Open the merge box (as Squash or Merge)
3. Change title (the message will appear)
4. Press `cancel` (the message will disappear)
5. Merge (the PR title will **not** be updated)

</details>

**Untested**

Where admin permissions are required

### CODE REVIEW

File: source/features/pr-merge-info-enhancements.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	wait: 1000,
+	immediate: true
+}) as () => string;
+
+function getPRNumber(): string {
+	return select('.gh-header-number')!.textContent!;
+}
+
+// Updates the field and dispatches the right events (focus is also used by `fit-textareas`)
+function updateField(selector: string, value: string): void {
```
Comment (notlmn) [inline source/features/pr-merge-info-enhancements.tsx]: Why explicit return type of `void`?

If required then why not at mentioned [here](https://github.com/sindresorhus/refined-github/pull/1934/files#diff-7aa792454424677c11856ea64b139ab9R69) too?

Comment (sindresorhus) [reply source/features/pr-merge-info-enhancements.tsx]: Explicit return type prevents bugs where you accidentally change the code inside the function to return a different type.

Comment (notlmn) [reply source/features/pr-merge-info-enhancements.tsx]: But `void` is not used for other functions, in the current feature and other features too.

Either they should be used everywhere, for all functions, or for none (implicit).

Comment (bfred-it) [reply source/features/pr-merge-info-enhancements.tsx]: It will be soon 

https://github.com/sindresorhus/refined-github/pull/1740#discussion_r249408719

**Comment by bfred-it** [comment]
I just tested the "update PR title" part when merging this: https://github.com/sindresorhus/refined-github/pull/1941 (I had initially changed the title manually and then remembered to try this featyre)

**Comment by bfred-it** [comment]
This is ready! I tested it a few times and it works correctly

**Comment by sindresorhus** [comment]
> Should we drop “fix-squash-and-merge-message”?

Sure. I'm kinda neutral. I don't find either the PR description (with RG) nor the list of commits (without RG) useful. The initial intention with the feature was good, in that it was more likely the PR description was useful than a list of commits. In reality, nobody writes a good PR description or it's too messy or verbose.

Ideally, it would just extract the `Fixes #1232` statements and add that to the merge commit description, as that's what I usually do manually.

> Does the "PR title update" part belong to the same feature?

Yes

**Comment by bfred-it** [comment]
> Should we drop “fix-squash-and-merge-message”?

Dropped in https://github.com/sindresorhus/refined-github/commit/9be9440699d2b8040af0c7c26ae524169b3cd0b8

**Comment by bfred-it** [comment]
Oops, merged prematurely while testing. Sent again as #1960

**Comment by zachwhaley** [comment]
Wait, is squash and merge _message_ being dropped?  We actually use this :(

**Comment by bharathwaaj** [comment]
We are also using this :( Please undo

**Comment by sindresorhus** [comment]
You can disable the feature in the extension settings.

**Comment by billdybas** [comment]
> In reality, nobody writes a good PR description or it's too messy or verbose.

@sindresorhus Since our team's process is _only_ to squash-and-merge, I know I take great care about writing good PR descriptions because I know the GitHub default of listing the commit names is not useful (since we all know the one commit that makes it into `master` really matters). 

I strongly urge you to restore the previous behavior of replacing the squash commit's information w/ the PR information. This was the primary reason why I and many of my teammates even considered using Refined GitHub, and now I'm not sure we even find any value in this extension anymore.

**Comment by yakov116** [comment]
@sindresorhus @bfred-it I happen to agree. I used this too.

**Pull request merged**